### ISSUE TITLE
Deprecating or modifying some tests?

### ISSUE THREAD

**Issue by bdarcus** [citation-style-language/test-suite#52]
@jgm earlier posted this over at the Discourse, noting some tests he found problematic.

https://discourse.citationstyles.org/t/upcoming-csl-meetup-context/1767/12

Any suggestions on what we do them?

The questions and commenting below are John's.

-----------------

Here are some examples of failures from citeproc-hs. In each case I may have simply missed a relative part of the spec.

Does the spec specify these behaviors for quotes?

```diff
[FAILED]   test/csl/flipflop_OrphanQuote.txt
--- expected
+++ actual
@@
-Nation of "Positive Obligations " of State under the European Convention on Human Rights (1)
+Nation of “Positive Obligations ” of State Under the European Convention on Human Rights (1)

[FAILED]   test/csl/quotes_QuotesUnderQuotesFalse.txt
--- expected
+++ actual
@@
 <div class="csl-bib-body">
-  <div class="csl-entry"> 'Title with ‘quotes’ in it',.</div>
+  <div class="csl-entry"> ’Title with ‘quotes’ in it’,.</div>
 </div>
```

Does the spec say to do this? (As noted in my last mail, it’s not always desirable.)

```diff
[FAILED]   test/csl/magic_CapitalizeFirstOccurringTerm.txt
--- expected
+++ actual
@@
-And
+and
```

Does the spec specify these things?

```diff
[FAILED]   test/csl/sort_OmittedBibRefMixedNumericStyle.txt
--- expected
+++ actual
@@
   <div class="csl-entry">1. Anderson, Book One</div>
-  <div class="csl-entry">2. [CSL STYLE ERROR: reference with no printed form.]</div>
+  <div class="csl-entry">[CSL STYLE ERROR: reference with no printed form.]</div>
   <div class="csl-entry">3. Crane, Book Two</div>

[FAILED]   test/csl/sort_OmittedBibRefNonNumericStyle.txt
--- expected
+++ actual
@@
   <div class="csl-entry">Anderson, Book One</div>
+  <div class="csl-entry">[CSL STYLE ERROR: reference with no printed form.]</div>
   <div class="csl-entry">Crane, Book Two</div>
```

Here I wonder if the absence of space between the Latin and Chinese names is really intended?

```diff
[FAILED]   test/csl/name_EtAlWithCombined.txt
--- expected
+++ actual
@@
   <div class="csl-entry">John Doe。</div>
-  <div class="csl-entry">——著，Ziggy Zither等點校。</div>
+  <div class="csl-entry">——著，Ziggy Zither 等點校。</div>
[Edit: added later] In this case I don’t understand why the space before Frinkle (which is present in the reference database) disappears:

[FAILED]   test/csl/sort_LeadingApostropheOnNameParticle.txt
--- expected
+++ actual
@@
   <div class="csl-entry">d’Wander, W</div>
-  <div class="csl-entry">de’ Frinkle, B</div>
+  <div class="csl-entry">de’Frinkle, B</div>
   <div class="csl-entry">in ’t Horvath, P A B</div>
```

**Comment by zepinglee** [comment]
> Here I wonder if the absence of space between the Latin and Chinese names is really intended?
> 
> ```diff
> [FAILED]   test/csl/name_EtAlWithCombined.txt
> --- expected
> +++ actual
> @@
>    <div class="csl-entry">John Doe。</div>
> -  <div class="csl-entry">——著，Ziggy Zither等點校。</div>
> +  <div class="csl-entry">——著，Ziggy Zither 等點校。</div>

There is no universal standard about the space character between Han and western characters but omitting it is one of the accepting styles. 

In typesetting, there is usually a 1/6 to 1/4 em space (width) between Han and western glyphs. Many softwares like Adobe Indesign, Microsoft Office, and TeX with appropriate packages can handle this automatically. However people do not form a consistent style of mixed Han and western in the input text characters. As far as I know, companies like Apple and Google tend to insert these space character in Chinese but omit them in Japanese (compare <https://www.apple.com.cn/> and <https://www.apple.com/jp/>).

**Comment by bdarcus** [comment]
So what do you suggest on that one, @zepinglee? Leave as is?

**Comment by zepinglee** [comment]
> So what do you suggest on that one, @zepinglee? Leave as is?
> 
> Aside: the others ones are all also, or maybe more, about the spec.

Yes. If I remember correctly, the output of `citeproc-js` is in this compact style.

**Comment by jgm** [comment]
Wouldn't it be sensible to have a set of conformance tests that only test behavior that is required by the spec?  (And not also behavior that is not spec but happens to have been implemented by citeproc.js?)  What's not controversial is that this issue is not in the spec.

**Comment by bdarcus** [comment]
Right, so really this is more a spec issue that anything? E.g. I should move this issue there?
